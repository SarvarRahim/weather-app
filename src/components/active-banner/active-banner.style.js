import styled from 'styled-components';

import ActiveBannerCardBackground from '../../assets/img/active-banner-bgi.png';

export const ActiveBannerCard = styled.div`
  padding: 24px 32px;
  background-image: url(${ActiveBannerCardBackground});
  background-repeat: no-repeat;
  background-size: cover;
  border-radius: 4px;
  overflow: hidden;
`;

export const ActiveBannerTitle = styled.h2`
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  line-height: 32px;
  letter-spacing: -0.02em;
  color: #000000;
  max-width: 208px;
  margin-bottom: 1.6rem;
`;

export const ActiveBannerSubtitle = styled.h4`
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #ff7a00;
  margin-bottom: 7px;
`;

export const ActiveBannerTime = styled.h3`
  font-style: normal;
  font-weight: bold;
  font-size: 32px;
  line-height: 32px;
  letter-spacing: -0.02em;
  color: #ff7a00;
  margin-bottom: 2rem;
`;

export const ActiveBannerLink = styled.div`
  a {
    font-style: normal;
    font-weight: bold;
    font-size: 16px;
    line-height: 22px;
    letter-spacing: -0.02em;
    color: #000000;
    text-decoration: none;
    display: block;

    svg {
      margin-left: 16px;
      transition: all ease 0.3s;
    }

    &:hover {
      svg {
        margin-left: 22px;
      }
    }
  }
`;
