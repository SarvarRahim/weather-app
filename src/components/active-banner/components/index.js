export * from './blurred-text.component';
