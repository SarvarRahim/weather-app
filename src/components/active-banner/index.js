export * from './active-banner.component';
