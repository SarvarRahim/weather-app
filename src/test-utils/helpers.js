import { render } from '@testing-library/react';
import React, { useRef } from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '@styles/Theme';
import { store } from '@store';
import { Formik } from 'formik';
import { VideoContext } from '@courses/routes/courses-single/course.context';
import { createMemoryHistory } from 'history';

const AllTheProviders = ({ children, formikValues = {} }) => {
  const videoRef = useRef();
  return (
    <BrowserRouter>
      <ReduxProvider store={store}>
        <ThemeProvider theme={theme}>
          <Formik initialValues={formikValues}>
            <VideoContext.Provider value={videoRef}>
              {children}
            </VideoContext.Provider>
          </Formik>
        </ThemeProvider>
      </ReduxProvider>
    </BrowserRouter>
  );
};

export const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};
