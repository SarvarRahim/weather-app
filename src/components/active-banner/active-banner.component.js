import { useEffect, useMemo, useState } from 'react';
import { hoc } from '@utils/hoc';
import { useDispatch, useSelector } from 'react-redux';
import Countdown, { zeroPad } from 'react-countdown';

import {
  ActiveBannerCard,
  ActiveBannerTitle,
  ActiveBannerSubtitle,
  ActiveBannerTime,
  ActiveBannerLink
} from './active-banner.style';
import { commonActions } from 'src/store/common/common.slice';
import { dateFormat } from '@utils/date-format';
import { BlurredText } from './components';

const useActiveBannerProps = () => {
  const [timeIndex, setTimeIndex] = useState(0);
  const coursePromoTimer = useSelector(state => state.common.promoCourses);

  const dispatch = useDispatch();

  const deadlineDate = useMemo(() => {
    if (!coursePromoTimer) return { iso: '', time: Date.now() };
    const date = new Date(parseInt(coursePromoTimer?.deadline) * 1000);
    setTimeIndex(timeIndex + 1);
    return {
      iso: dateFormat(date),
      time: date.getTime()
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [coursePromoTimer]);

  useEffect(() => {
    dispatch(commonActions.fetchPromoCourses());
  }, [dispatch]);

  return {
    coursePromoTimer,
    deadlineDate,
    timeIndex
  };
};

export const ActiveBanner = hoc(
  useActiveBannerProps,
  ({ coursePromoTimer, deadlineDate, timeIndex, promocode }) => {
    if (!coursePromoTimer) {
      return null;
    }

    return (
      <ActiveBannerCard>
        <ActiveBannerTitle data-testid='active-banner-title' role='heading'>
          {coursePromoTimer?.title}
        </ActiveBannerTitle>

        <ActiveBannerSubtitle
          data-testid='active-banner-subtitle'
          role='heading'
        >
          Tugash vaqti: {deadlineDate.iso}
        </ActiveBannerSubtitle>

        <Countdown
          date={deadlineDate.time}
          key={timeIndex}
          renderer={props => {
            return (
              <ActiveBannerTime>
                {props.days} kun {zeroPad(props.hours)}:{zeroPad(props.minutes)}
                :{zeroPad(props.seconds)}
              </ActiveBannerTime>
            );
          }}
        />

        <ActiveBannerLink>
          <BlurredText>{coursePromoTimer?.serial || 'ac17b1a'}</BlurredText>
        </ActiveBannerLink>
      </ActiveBannerCard>
    );
  }
);
