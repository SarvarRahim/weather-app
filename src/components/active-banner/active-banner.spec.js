import { customRender } from '@test-utils/helpers';
import { screen } from '@testing-library/react';

import { ActiveBanner } from './active-banner.component';

const setup = ({ coursePromoTimer, deadlineDate, timeIndex }) => {
  return customRender(
    <ActiveBanner
      coursePromoTimer={coursePromoTimer}
      deadlineDate={deadlineDate}
      timeIndex={timeIndex}
    />
  );
};

test('Active banner test', () => {
  const { asFragment } = setup({
    coursePromoTimer: { title: 'tile', serial: 'serial' },
    deadlineDate: '15.25.54',
    timeIndex: '1'
  });
  expect(asFragment).toMatchSnapshot();
});

test('Active banner props test', () => {
  setup({
    coursePromoTimer: 'title',
    deadlineDate: '13.02.03',
    timeIndex: '1'
  });
  const activeBanenrDeadlineTime = screen.getByTestId('active-banner-subtitle');
  const activeBannerTitle = screen.getByTestId('active-banner-title');
  expect(activeBanenrDeadlineTime && activeBannerTitle).toBeTruthy();
});
