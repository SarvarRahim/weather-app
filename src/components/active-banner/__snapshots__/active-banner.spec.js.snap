// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active banner test 1`] = `[Function]`;
